synedrion/src/cggmp21/sigma/enc.rs

```diff
@@ -30,36 +29,36 @@ Public inputs:
 */
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub(crate) struct EncProof<P: SchemeParams> {
-    e: Signed<<P::Paillier as PaillierParams>::Uint>,
+    e: PublicSigned<<P::Paillier as PaillierParams>::Uint>,
     cap_s: RPCommitmentWire<P::Paillier>,
     cap_a: CiphertextWire<P::Paillier>,
     cap_c: RPCommitmentWire<P::Paillier>,
-    z1: Signed<<P::Paillier as PaillierParams>::Uint>,
+    z1: PublicSigned<<P::Paillier as PaillierParams>::Uint>,
     z2: MaskedRandomizer<P::Paillier>,
-    z3: Signed<<P::Paillier as PaillierParams>::WideUint>,
+    z3: PublicSigned<<P::Paillier as PaillierParams>::WideUint>,
 }
 
 impl<P: SchemeParams> EncProof<P> {
     pub fn new(
         rng: &mut impl CryptoRngCore,
-        k: &Secret<Signed<<P::Paillier as PaillierParams>::Uint>>,
+        k: &SecretSigned<<P::Paillier as PaillierParams>::Uint>,
         rho: &Randomizer<P::Paillier>,
         pk0: &PublicKeyPaillier<P::Paillier>,
         cap_k: &Ciphertext<P::Paillier>,
         setup: &RPParams<P::Paillier>,
         aux: &impl Hashable,
     ) -> Self {
-        k.expose_secret().assert_bound(P::L_BOUND);
+        k.assert_exp_range(P::L_BOUND);
```

Comment:
I renamed them because I didn't want confusion with the `bound` field of Signed. Maybe expand it to "exponent"? And "exp" is there to signify that it's the range of `2^x`, not `x`. Although maybe it's not that important since it's pretty much always that.

Reply:
I think "exponent" is better.

Reply:
Renamed